Readly/Networking/RequestService.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  RequestService.swift
+//  Readly
+//
+//  Created by Viorel Rinja on 4/24/21.
+//
+
+import Foundation
+import Alamofire
+import ObjectMapper
+
+class RequestService {
+    
+    typealias RequestResultCompletion = ((Data?, Int?) -> Void)?
+    typealias RequestServiceResult<T: BaseMappable> = (([T]?) -> Void)?
+    
+    static func retrieveData(_ url: String, completion: RequestResultCompletion) {
+        AF.request(url,
+                   method: .get)
+            .validate()
+            .response { (response) in
+                
+                let statusCode = response.response?.statusCode
+                completion?(response.data, statusCode)
+            }
+    }
+    
+    static func retrieveData<T: BaseMappable>(_ url: String, resultType: T.Type, completion: RequestServiceResult<T>)
+    {
+        AF.request(url,
+                   method: .get)
+            .validate()
+            .responseJSON { (response: AFDataResponse<Any>) in
+                
+                switch response.result
+                {
+                case .success(let json):
+                    completion?(Mapper<T>().mapArray(JSONObject: json))
+                default:
+                    completion?(nil)
+                }
+            }
+    }
+}
+
```

Comment:
@viorelrj Please refactor your `RequestService` to use [RxSwift](https://github.com/ReactiveX/RxSwift) in that way you will not use completion handlers which is not most convenient way of handling asynchronous events.
Example of how to do this you can also find in [sample iOS app](https://github.com/ASV44/sample-ios-app)